src/app/Form/FormScores/FormScores.js

```diff
@@ -0,0 +1,170 @@
+    isSaveButtonShown: PropTypes.bool.isRequired,
+    updateDirtyInputs: PropTypes.func.isRequired,
+    handleChange: PropTypes.func.isRequired,
+    showErrorMessage: PropTypes.func.isRequired,
+    handleSubmit: PropTypes.func.isRequired,
+    handleCancel: PropTypes.func.isRequired,
+}
+
+export default function FormScores({
+    formInputs,
+    isSaveButtonShown,
+    updateDirtyInputs,
+    handleChange,
+    showErrorMessage,
+    handleSubmit,
+    handleCancel}) {
+
+    return (
+        <Form onSubmit={handleSubmit}>
+            <Headline>Score</Headline>
+            <ScoreOverview>
+                <Legend>
+                    <span>Holes</span>
+                    {new Array(18).fill().map((_, index) => {
+                        const newId = uuid()
+                        return <span key={newId}>{index + 1}</span>
+                        })
+                    }
+                </Legend> 
+                <ScoresAllPlayers>
+                    {formInputs.playersArray?.map((player, index) => {
+                        const playerName = player
+                        const key = `singlePlayerInputs${index + 1}`    
+                            return (
+                                <ScoreSinglePlayer key={key} >
+                                    <span>{playerName}</span>
+                                    {new Array(18).fill().map((_, index) => {
+                                        const inputNumber = index + 1
+                                        const holeName = 'hole' + inputNumber
+                                        const key = `singleScoreInput${inputNumber}`
+                                            return( 
+                                                    <SingleScore key={key}>
+                                                        <input 
+                                                        type="number" 
+                                                        name={holeName + '-' + playerName}
+                                                        id={holeName + '-' + playerName}
+                                                        value={formInputs.scores[playerName][holeName]}
+                                                        onChange={(event) => {handleChange(event.target.name, event.target.value)}}
+                                                        onBlur={(event) => updateDirtyInputs(event.target.name)}
+                                                        />
+                                                    </SingleScore>
+                                                )                         
+                                        })
+                                    }
+                                </ScoreSinglePlayer>
+                            )
+                    })}
+                </ScoresAllPlayers> 
+            </ScoreOverview> 
+            <Note>{showErrorMessage()}</Note>
```

Comment:
Is it correct to execute the function here directly?

Reply:
It is intended that this function is directly executed, as it continuously checks, whether the inputs are valid and dirty. By doing this, it returns the error message at the right moment.